docs/source/upgrade.rst

```diff
@@ -4,6 +4,77 @@ Upgrade Guide
 
 This is a list of changes introduced in each release that users should be aware of when migrating from older versions.
 
+CuPy v10
+========
+
+Dropping Support of CUDA 9.2
+----------------------------
+
+CUDA 9.2 is no longer supported.
+Use CUDA 10.0 or later.
+
+Dropping Support of NCCL v2.4
+-----------------------------
+
+NCCL v2.4 is no longer supported.
+
+Changes in :class:`cupy.cuda.Stream` Behavior
+---------------------------------------------
+
+Stream is now managed per-device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Previoulys, it was users' responsibility to keep the current stream consistent with the current CUDA device.
+For example, the following code raises an error in CuPy v9 or earlier:
+
+.. code-block:: py
+
+   import cupy
+
+   with cupy.cuda.Device(0):
+       # Create a stream on device 0.
+       s0 = cupy.cuda.Stream()
+
+   with cupy.cuda.Device(1):
+       with s0:
+           # Try to use the stream on device 1
+           cupy.arange(10)  # -> CUDA_ERROR_INVALID_HANDLE: invalid resource handle
+
+CuPy v10 manages the current stream per-device so that you don't have to take care of the consistency between streams and devices.
+In the above example, in CuPy v10 ``with s0:`` only changes the current stream for the device 0, so ``cupy.arange(10)`` will run on the current stream for device 1, which is by default the legacy (null) default stream.
```

Comment:
I feel this needs to be a bit more detailed, it was hard for me to grasp the complete meaning.
How about something like

```
When using CuPy v10, the above example behaves differently because whenever a stream is created, it is automatically associated with the current device and will be ignored when switching devices. 
In early versions, trying to use `s0` in device 1 crashes because `s0` is associated with device 0. However, in v10, this `s0` is ignored and the default stream for device 1 will be used instead.
```

Reply:
```suggestion
When using CuPy v10, the above example behaves differently because whenever a stream is created, it is automatically associated with the current device and will be ignored when switching devices. 
In early versions, trying to use `s0` in device 1 raises an error because `s0` is associated with device 0. However, in v10, this `s0` is ignored and the default stream for device 1 will be used instead.
```